src/app/pages/api/openai.ts

```diff
@@ -4,15 +4,28 @@ import { OpenaiApiType } from '@/app/types/OpenaiApiType';
 
 dotenv.config({ path: path.resolve(__dirname, '../..', '.env.local') });
 
-const fetchData = async (): Promise<OpenaiApiType> => {
+type getChatResponseType = {
+  knowledge: string[];
+};
+
+const getChatResponse = async (
+  props: getChatResponseType
+): Promise<OpenaiApiType> => {
   // source: https://supabase.com/docs/guides/getting-started/openai/vector-search
   const temperamentPrimer =
     'You are a helpdesk assistant for WizzAir who loves to help users find the correct information for their questions from our FAQs.';
   const knowledgeBasePrimer =
     'If you are unsure and the answer is not explicitly written in the information provided, say "Sorry, I don\'t know how to help with that.';
   const legalityPrimer = 'You should not provide legal advice of any kind.';
+
   const userDummyMessage = 'This is a test message!';
 
+  const customKnowledge = props.knowledge
```

Comment:
Can maybe add a type for these {role, content} objects?

Reply:
good shout!